### DISCUSSION TITLE
error while passing props / Maximum call stack size exceeded

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#47612] [Help] [answered]
### Summary

Hello! I am using the app directory.

In the server component, I make a request to mongodb:
`const user = await User.findOne({name:'Alex'})`
It works well. I can get information about the 'user' in the console.
But if I try to pass 'user' to another component, it's an error. How can I fix this?

error - Maximum call stack size exceeded

if i pass any other variable (not from mingodb) there is no error.
Error in server console and client console.

### Additional information

```js
next: 13.2.1
```

**Comment by icyJoseph** [comment, marked as answer]
This might be related to: https://flaviocopes.com/nextjs-serialize-date-json/

IIRC the returned object from that query is not just a `key-value` pair dictionary, but a bit more complex JavaScript object (aka not a plain object), so you gotta serialize it first.

**Comment by thiagomurtinho** [comment]
I recommend reading the library **next-mdx-remote**

[Serialize example](https://www.npmjs.com/package/next-mdx-remote#Examples)

Your code should look like this:
```
import { serialize } from 'next-mdx-remote/serialize'
import { MDXRemote } from 'next-mdx-remote'

import Test from '../components/test'

const components = { Test }

export default function TestPage({ source }) {
  return (
    <div className="wrapper">
      <MDXRemote {...source} components={components} />
    </div>
  )
}

export async function getStaticProps() {
  // MDX text - can be from a local file, database, anywhere
  const source = 'Some **mdx** text, with a component <Test />'
  const mdxSource = await serialize(source)
  return { props: { source: mdxSource } }
}
```
